pandas/core/internals/construction.py

```diff
@@ -682,7 +704,7 @@ def _validate_or_indexify_columns(
 
 
 def _convert_object_array(
-    content: List[Scalar], coerce_float: bool = False, dtype: Optional[Dtype] = None
+    content: List[Scalar], coerce_float: bool = False, dtype: Optional[DtypeObj] = None
```

Comment:
would it make sense to change L724 to an isinstance check if the allow parameter types are restricted.

Reply:
i dont think so.  we could change the comparison to `dtype != np.dtype("O")`